### PR TITLE
Exclude certain Pods from Acknowledgements

### LINKED ISSUE #16: Exclude certain Pods from Acknowledgements
Without the ability for Podspecs to opt-out of this stuff, we need some way to remove private pods and other pods that don't require attribution.

Proposal:

```ruby
plugin 'cocoapods-acknowledgements', :settings_bundle => true, :exclude => 'PrivateKit'
```

```ruby
plugin 'cocoapods-acknowledgements', :settings_bundle => true, :exclude => ['PrivateKit', 'SecretLib']
```

### PR DESCRIPTION
**Pull request by marcelofabri**
Fixes #16.

### CODE REVIEW

**Comment by marcelofabri** [comment]
@orta Have you had a chance to look at this yet?

File: lib/cocoapods_acknowledgements/plist_generator.rb [hunk 1]
```diff
@@ -8,8 +8,8 @@ def markdown_parser
         @markdown_parser ||= Redcarpet::Markdown.new(Redcarpet::Render::HTML)
       end
 
-      def generate(target_description, sandbox)
-        root_specs = target_description.specs.map(&:root).uniq
+      def generate(target_description, sandbox, excluded)
+        root_specs = target_description.specs.map(&:root).uniq.select { |spec| !excluded.include? spec.name }
```
Comment (segiddins) [inline lib/cocoapods_acknowledgements/plist_generator.rb]: make this `reject { |spec| excluded.include?(spec.name) }`

File: lib/cocoapods_acknowledgements/settings_plist_generator.rb [hunk 2]
```diff
@@ -4,8 +4,8 @@ module CocoaPodsAcknowledgements
   class SettingsPlistGenerator < PlistGenerator
     class << self
 
-      def generate(target_description, sandbox)
-        root_specs = target_description.specs.map(&:root).uniq
+      def generate(target_description, sandbox, excluded)
+        root_specs = target_description.specs.map(&:root).uniq.select { |spec| !excluded.include? spec.name }
```
Comment (segiddins) [inline lib/cocoapods_acknowledgements/settings_plist_generator.rb]: same as above

File: lib/cocoapods_acknowledgements.rb [hunk 3]
```diff
@@ -46,6 +46,7 @@ def self.settings_bundle_in_project
     Pod::UI.section 'Adding Acknowledgements' do
 
       should_include_settings = user_options["settings_bundle"] != nil
+      excluded_pods = Array(user_options["excluded"])
```
Comment (segiddins) [inline lib/cocoapods_acknowledgements.rb]: This should probably be `Set.new(user_options["excluded"])`, which would require  a `require 'set'` in here

**Comment by marcelofabri** [comment]
@segiddins I've addressed your comments. Thanks for taking a look at this :+1:

**Pull request merged**